File: directions-reduction/directions-reduction.py
def dirReduc(arr):
    l  = len(arr)
    opp = {"NORTH":"SOUTH", "SOUTH":"NORTH","EAST":"WEST","WEST":"EAST"}
    j = 0 
    while j < l/2:
        for i in range(max(len(arr)-1,0)):
            if i+1<len(arr) and arr[i] == opp[arr[i+1]]:
                arr = arr[:i] + arr[i+2:]
        j += 1
    return arr

File: directions-reduction/directions-reduction.js
function dirReduc(arr){
  let s = arr.join("-")
  let i =0;
  while(i<Math.floor(arr.length/2)){
  s = s.replace(/NORTH-SOUTH|SOUTH-NORTH|EAST-WEST|WEST-EAST/g,"").replace(/--/,"-").replace(/^-|-$/,"")
  i+=1}
  return s!==""? s.split("-") : []
}